app-modules/student-data-model/tests/Tenant/Student/Filament/Resources/StudentResource/RelationManagers/InteractionsRelationManagerTest.php

```diff
@@ -0,0 +1,221 @@
+
+    $user->givePermissionTo('student.view-any');
+    $user->givePermissionTo('student.*.view');
+    $user->givePermissionTo('interaction.view-any');
+
+    actingAs($user);
+
+    livewire(InteractionsRelationManager::class, [
+        'ownerRecord' => $student,
+        'pageClass' => ViewStudent::class,
+    ])
+        ->filterTable('interaction_type_id', $types->first()->id)
+        ->assertCanSeeTableRecords($student->interactions->where('interaction_type_id', $types->first()->id))
+        ->assertCanNotSeeTableRecords($student->interactions)
+        ->removeTableFilter('interaction_type_id')
+        ->assertCanSeeTableRecords($student->interactions)
+        ->assertCanNotSeeTableRecords($student->interactions->where('interaction_type_id', $types->first()->id));
+});
+
+it('renders the status select filter', function () {
+    $user = User::factory()->licensed(Student::getLicenseType())->create();
+
+    $student = Student::factory()
+        ->has(Interaction::factory()->count(5))
+        ->create();
+
+    $statuses = InteractionStatus::factory()->count(5)->create();
+
+    $user->givePermissionTo('student.view-any');
+    $user->givePermissionTo('student.*.view');
+    $user->givePermissionTo('interaction.view-any');
+
+    actingAs($user);
+
+    livewire(InteractionsRelationManager::class, [
+        'ownerRecord' => $student,
+        'pageClass' => ViewStudent::class,
+    ])
+        ->filterTable('interaction_status_id', $statuses->first()->id)
+        ->assertCanSeeTableRecords($student->interactions->where('interaction_status_id', $statuses->first()->id))
+        ->assertCanNotSeeTableRecords($student->interactions)
+        ->removeTableFilter('interaction_status_id')
+        ->assertCanSeeTableRecords($student->interactions)
+        ->assertCanNotSeeTableRecords($student->interactions->where('interaction_status_id', $statuses->first()->id));
+});
+
+it('renders the created by select filter', function () {
+    $user = User::factory()->licensed(Student::getLicenseType())->create();
+
+    $student = Student::factory()
+        ->has(Interaction::factory()->count(5))
+        ->create();
+
+    $users = User::factory()->count(5)->create();
+
+    $user->givePermissionTo('student.view-any');
+    $user->givePermissionTo('student.*.view');
+    $user->givePermissionTo('interaction.view-any');
+
+    actingAs($user);
+
+    livewire(InteractionsRelationManager::class, [
+        'ownerRecord' => $student,
+        'pageClass' => ViewStudent::class,
+    ])
+        ->filterTable('user_id', $users->first()->id)
+        ->assertCanSeeTableRecords($student->interactions->where('user_id', $users->first()->id))
+        ->assertCanNotSeeTableRecords($student->interactions)
+        ->removeTableFilter('user_id')
+        ->assertCanSeeTableRecords($student->interactions)
+        ->assertCanNotSeeTableRecords($student->interactions->where('user_id', $users->first()->id));
+});
```

Comment:
These filter tests are constructed in a way that the randomness makes them a little rough in whether or not they are actually testing that the filter is working as it should.

You should be a little more explicit.

Taking the status one as an example. Create 2 or more `InteractionStatus`es first. Then create `Interaction`/s for the Student with each manually, then use the filter and assert you see some and not the others.